synapse/handlers/register.py

```diff
@@ -275,16 +275,12 @@ def _auto_join_rooms(self, user_id):
         fake_requester = create_requester(user_id)
 
         # try to create the room if we're the first real user on the server. Note
-        # that an auto-generated support user is not a real user and will never be
+        # that an auto-generated support or bot user is not a real user and will never be
         # the user to create the room
         should_auto_create_rooms = False
-        is_support = yield self.store.is_support_user(user_id)
-        # There is an edge case where the first user is the support user, then
-        # the room is never created, though this seems unlikely and
-        # recoverable from given the support user being involved in the first
-        # place.
-        if self.hs.config.autocreate_auto_join_rooms and not is_support:
-            count = yield self.store.count_all_users()
+        is_real_user = yield self.store.is_real_user(user_id)
+        if self.hs.config.autocreate_auto_join_rooms and is_real_user:
+            count = yield self.store.count_real_users()
```

Comment:
I wonder whether it'd be easier to just check if the alias exists, rather than trying to count?

Reply:
So the reason for the count, AFAICT, is that we only want to auto-create rooms when the first (real) user registers. I was initially going to suggest a change that rooms would always be auto-created if missing, but the problem there is that the next registering user would own any room added to the Synapse configuration.

Ideally rooms would be created on Synapse start-up to avoid this. But then the problem is who do you make the creator?

Reply:
Hum, true. It sort of feels like we're using the "first real user" as a proxy for whether that user is an admin or not. I wonder if we should change it to check for the admin bit? I dunno.

Reply:
/me throws the "I agree things could be better but since this is modifying existing functionality and things are generally broken for our use case it might be good to fix first, then refactor" -card out there ;)

Reply:
But seriously, one possible solution would be to simplify things to not auto-create rooms, making admins who *configure auto-join* to pre-create the rooms. This would be simplest and avoid the problems and would probably not be too much to ask for admins to do.

Reply:
Or in fact have a configuration option for a "homeserver" account, e.g. `@admin:example.com` which gets used to create the room

Reply:
When this was first implemented, using the first user was considered a 'good enough' hack to get going, definitely could be improved.